### ISSUE TITLE
VITON-HD: High-Resolution Virtual Try-On via Misalignment-Aware Normalization

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#164]
## TL; DR

- Provide additional segmentation map for person representation with the *clothing* and *arm* removed.
- First predict target segmentation map with segmentation generator.
- Use predicted segmentation map with ALIignment Aware Segment Normalization layers to minimize misalignment.

## Motivation

- Dilemma of clothing-agnostic person representation
- (1) VITON and CP-VTON uses down-sampled person shape (segmentation map) which makes the body shape blurry
- (2) ACGPN uses accurate body shape mask, which leaks the clothing information (such as shape), hence degrading generalization performance.

## Prior works

- VITON
- CP-VTON
- ACGPN

## Method

![image](https://user-images.githubusercontent.com/46308407/160385719-b136a815-fb4f-4c7d-ac24-e45e42a1e5b8.png)

### Clothing-Agnostic Person Representation

- With only target clothing image and the GT reference image, we should train a ViTON model that can generalize to unseen clothing.
- In order to do that, we should eliminate the clothing information while preserving the person information.
- To this end, first obtain semantic segmentation map as well as 3D RGB pose map from the reference image.
- Then, erase target clothing area as well as arms (*easily susceptible to occlusions*) using segmentation map and pose map, respectively.
- Additionally generate clothing agnostic reference image where clothing and arm area is *erased*.
- As a result, the clothing-agnostic person representation consists of clothing-agnostic segmentation map, RGB pose map, and clothing-agnostic image.

### Segmentation Generation

- Input: clothing-agnostic segmentation map + pose map + original clothing image
- Output: predicted segmentation map of the reference person wearing the target clothing
- Objective: pixel-wise cross-entropy loss as well as conditional adversarial loss

### Clothing Deformation

- Geometric Matching Module (GMM) in CP-VTON is used.
- A = Features from concat(clothing-agnostic image, posemap, clothing area mask from predicted segmentation map)
- B = Features from original clothing image
- Correlation matrix Corr(A, B) is fed to a regression network that predicts TPS transformation parameters (*2x5x5=50-dim*)
- TPS warps the original clothing to the desired pose and shape.
- L1 loss and 2nd derivative smoothness prior is imposed.

### ALIAS Normalization

![image](https://user-images.githubusercontent.com/46308407/160385745-d1cfb58b-218c-49a4-ab1e-c036ec244980.png)
![image](https://user-images.githubusercontent.com/46308407/160385765-599dd1d4-5dc4-45c2-8f43-0138a822e629.png)

- First generate misalignment mask by subtracting TPS-warped mask from target clothing mask obtained from predicted segmentation map.
- Separately standardize the features according to the misalignment mask, then apply learned modulation.
- Modulation parameters are predicted from convolutional layers taking clothing masks and segmentation map.

## Experiments

![image](https://user-images.githubusercontent.com/46308407/160386533-a05ed5a9-83cf-4cdf-878f-7a7a341516e6.png)
![image](https://user-images.githubusercontent.com/46308407/160386560-122477cd-f8ce-4a70-bd1f-a6f7c121a398.png)
![image](https://user-images.githubusercontent.com/46308407/160386580-42e4e247-748d-408a-8593-05b89c7baaf9.png)
![image](https://user-images.githubusercontent.com/46308407/160386609-f412f69d-33cf-4428-9853-857caa2c8e45.png)

## Comments

- Nice intuition of using segmentation map as representation, where clothing and *arm* part can be effectively removed. (*while keeping the hands, which is difficult to reconstruct*)
